lib/ansible/module_utils/basic.py

```diff
@@ -1611,7 +1611,12 @@ def _check_type_float(self, value):
 
     def _check_type_path(self, value):
         value = self._check_type_str(value)
-        return os.path.expanduser(os.path.expandvars(value))
+        valid_path = os.path.expanduser(os.path.expandvars(value))
+
+        if valid_path == '':
```

Comment:
I have proven in the PR description that it is in deed invalid.

If it doesn't hit this function with `None` value then we don't need to test the `None` here at all.